File: BulbShare/ios/RNBulbShare.m

//#import "RNBulbShare.h"
//
//@implementation RNBulbShare
//
//- (dispatch_queue_t)methodQueue
//{
//    return dispatch_get_main_queue();
//}
//RCT_EXPORT_MODULE()
//
//@end
  
#import <React/RCTBridgeModule.h>
#import <Foundation/Foundation.h>


@interface RCT_EXTERN_MODULE(RNBulbShare, NSObject)
RCT_EXTERN_METHOD(
                  initSdk: (RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject
                  )
@end

File: BulbShare/ios/RNBulbShare.swift
//
//  RNBulbShare.swift
//  RNBulbShare
//
//  Created by sanchit on 05/01/22.
//  Copyright © 2022 Facebook. All rights reserved.
//

import Foundation
import UIKit

@objc(RNBulbShare)
class RNBulbShare: NSObject {
  @objc static func requiresMainQueueSetup() -> Bool {
      return false
  }
  
  @objc func initSdk(_ resolve:  @escaping RCTPromiseResolveBlock,rejecter reject:  @escaping RCTPromiseRejectBlock) -> Void{
    DispatchQueue.main.async {
        print("HIIII")
      resolve("true")
    }
  }
}
